### PR TITLE
DocBook. Full-Stack Capstone Project.

### PR DESCRIPTION
**Pull request by CaptainBawa**
### In this project:

- [ ] We completed the following:
     - Built the backend with Rails and PostgreSQL as the database.
     - Used devise and cancan gem for authorization and authentication.
     - Created models and controllers for the app.
     - Wrote tests for all the models and controllers.
     - Deployed the backend on render.
- [ ] Frontend PR link: (https://github.com/CaptainBawa/DocBook-Front-End/pull/15)

### CODE REVIEW

File: app/controllers/users/confirmations_controller.rb
Comment (Meltrust) [inline app/controllers/users/confirmations_controller.rb:1]: - [ ] Kindly, remove the old commented code present in this file.  That way, we will improve the repo's cleanliness. 👍

File: app/controllers/users/omniauth_callbacks_controller.rb
Comment (Meltrust) [inline app/controllers/users/omniauth_callbacks_controller.rb:1]: - [ ] Kindly, remove the old commented code present in this file.  That way, we will improve the repo's cleanliness. 👍

File: app/controllers/users/passwords_controller.rb
Comment (Meltrust) [inline app/controllers/users/passwords_controller.rb:1]: - [ ] Kindly, remove the old commented code present in this file.  That way, we will improve the repo's cleanliness. 👍

File: app/controllers/users/unlocks_controller.rb
Comment (Meltrust) [inline app/controllers/users/unlocks_controller.rb:1]: - [ ] Kindly, remove the old commented code present in this file.  That way, we will improve the repo's cleanliness. 👍

File: README.md [hunk 1]
```diff
@@ -1,6 +1,4 @@
-<<<<<<< HEAD
-# DocBook-Back-End
-=======
+
```
Comment (Meltrust) [inline README.md:1]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

File: app/controllers/appointments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      render json: @appointment, include: %i[user doctor]
+    else
+      render json: { error: @appointment.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @appointment.destroy
+    head :no_content
+  end
```
Comment (Meltrust) [inline app/controllers/appointments_controller.rb]: - [ ] What happens if the appointment is not successfully destroyed?  Kindly add some error handling here. 👍

File: app/controllers/doctors_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    user = User.find(params[:user_id])
+    doctor = user.doctors.find(params[:id])
+    render json: doctor, status: :ok
+  end
+
+  def destroy
+    doctor = Doctor.find(params[:id])
+    doctor.destroy
+    head :no_content
+  end
```
Comment (Meltrust) [inline app/controllers/doctors_controller.rb]: - [ ] What happens if the doctor is not successfully destroyed?  Kindly add some error handling here. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @CaptainBawa,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Comment by CaptainBawa** [comment]
### Dear Reviewer 

We've completed all the requested modifications as per the previous reviewer's feedback in the backend repository. Additionally, we've also provided comments in the frontend repository.

**Review by mcrd25 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Made requested changes 👍 
## Optional suggestions
- Consider being consistent with Error Handling
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Hope all is well, and that my comments were helpful. 😸**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_Also, you do not have to implement all requested changes, as this is up to you in the end, but I strongly recommend you implement them._

**Pull request merged**